### PR TITLE
Add platform type/build to info sent to register

### PR DESCRIPTION
**Pull request by penrods**
Add platform information (type and build #) to the messages
send to the server to pair and to update version info.

### CODE REVIEW

**Comment by penrods** [comment]
@augustnmonteiro and @matheuslima let me know if this looks correct.  I had it sitting on my local machine and wanted to get the code off of it for review or trash.

**Comment by penrods** [comment]
I should expand on this -- there should be no reason to set the device type manually in home.mycroft.ai when you register a device.  This change is intended to get the information to the server, and obviously the server will have to listen for it.  Does this change make sense?

File: mycroft/api/__init__.py [hunk 1]
```diff
@@ -144,21 +144,35 @@ def get_code(self, state):
...
+        platform = "unknown"
+        platform_build = ""
+        if "enclosure" in config:
+            platform = config.get("enclosure").get("platform", "unknown")
+            platform_build = config.get("enclosure").get("platform_build", "")
         return self.request({
             "method": "PATCH",
             "path": "/" + self.identity.uuid,
             "json": {"coreVersion": version.get("coreVersion"),
+                     "platform": platform,
```
Comment (augustnmonteiro) [inline mycroft/api/__init__.py]: the platform is the called type on the backend so if should be
`"type": platform,`

File: mycroft/api/__init__.py [hunk 2]
```diff
@@ -144,21 +144,35 @@ def get_code(self, state):
...
+                     "platform_build": platform_build,
                      "enclosureVersion": version.get("enclosureVersion")}
         })
 
     def update_version(self):
         version = VersionManager.get()
+        platform = "unknown"
+        platform_build = ""
+        if "enclosure" in config:
+            platform = config.get("enclosure").get("platform", "unknown")
```
Comment (augustnmonteiro) [inline mycroft/api/__init__.py]: the platforms is a enum with the types `Mycroft, Desktop, Picroft, Other`

File: mycroft/api/__init__.py [hunk 3]
```diff
@@ -144,21 +144,35 @@ def get_code(self, state):
...
+            platform_build = config.get("enclosure").get("platform_build", "")
         return self.request({
             "method": "PATCH",
             "path": "/" + self.identity.uuid,
             "json": {"coreVersion": version.get("coreVersion"),
+                     "platform": platform,
+                     "platform_build": platform_build,
```
Comment (augustnmonteiro) [inline mycroft/api/__init__.py]: we don't have this field on the backend

Comment (penrods) [reply mycroft/api/__init__.py]: Let's add it!

**Comment by penrods** [comment]
If we are changing the backend to eliminate "type" as an enum, then we can go ahead and leave this as a new string named "platform" and "platform_build".  This will be ignored today, but once the backend has made this migration it will start picking up the new values.

**Comment by forslund** [comment]
@penrods I think the consensus was that this should be merged and the backend behavior should be updated accordingly.

There is however an error  (config is undeclared when used to get the platform type), if you'll fix that I'll merge this

**Comment by penrods** [comment]
Fixed!

**Comment by forslund** [comment]
Looks good. Merging

**Pull request merged**